crates/shared/src/price_estimation/oneinch.rs

```diff
@@ -72,12 +71,11 @@ impl PriceEstimating for OneInchPriceEstimator {
             a SanitizedPriceEstimator"
         );
 
-        future::join_all(
-            queries
-                .iter()
-                .map(|query| async move { self.estimate(query).await }),
-        )
-        .await
+        let mut results = Vec::new();
```

Comment:
nit: This could allocate the correct size immediately: `Vec::with_capacity(queries.len())`